Fix anchor tag issues

gusseting opened the pull request:
Moved anchor tags above headings to fix display issues.
Before:
![chocprob1](https://cloud.githubusercontent.com/assets/540264/17995219/4da97ace-6ba3-11e6-9ca5-d8c9c0292190.png)

After:
![chocprob2](https://cloud.githubusercontent.com/assets/540264/17995222/5a308da0-6ba3-11e6-9ea8-2a15b919c2b3.png)

As noted in https://github.com/exercism/exercism.io/issues/3078, the fix doesn't completely fix display issues. I'm working on the main issue for the CLI - https://github.com/exercism/exercism.io/issues/2535 and hope to have a pull request if not within hours, by early next week which has more major changes to the copy and layout.
This fix should resolve #3078

Insti commented on x/docs/md/cli/mac.md:
```diff
@@ -25,7 +25,9 @@ Homebrew/homebrew-core (git revision 3b4c; last commit 2016-05-09)
 
-### Installing With Homebrew <a name="installing-with-homebrew"></a>
+
+<a name="installing-with-homebrew"></a>
```
This is a markdown file.
Markdown has it's own syntax for adding anchors and one should not mix html/markdown formatting.
Since the anchor is the same as the heading, The anchors to the headings should be created by whatever parses the markdown.

kytrinyx replied:
Does markdown know how to make "name" anchors, though?

Insti replied:
From https://gist.github.com/asabaylus/3071099
> To create an anchor to a heading in github flavored markdown. Add - characters between each word in the heading and wrap the value in parens (#some-markdown-heading) so your link should look like so:
> 
> `[create an anchor](#anchors-in-markdown)`

gusseting commented:
@insti @kytrinyx 
I've tried the above suggestion locally, and the link doesn't work, I suspect because the syntax: '[create an anchor](#anchors-in-markdown)' is _github flavoured markdown_ and we have _exercism flavoured markdown_.

I haven't written any new code here, merely moved the old code in order to fix the viewing issue. The old code was also written by me, and previously reviewed in this pull request:
https://github.com/exercism/exercism.io/pull/2874
I'm hoping that given the below from the founder of markdown that we can accept the code that is there, which works, and is temporary.

> Markdown is not a replacement for HTML, or even close to it. Its syntax is very small, corresponding only to a very small subset of HTML tags. The idea is not to create a syntax that makes it easier to insert HTML tags. In my opinion, HTML tags are already easy to insert. The idea for Markdown is to make it easy to read, write, and edit prose. HTML is a publishing format; Markdown is a writing format. Thus, Markdown’s formatting syntax only addresses issues that can be conveyed in plain text.
> 
> For any markup that is not covered by Markdown’s syntax, you simply use HTML itself. There’s no need to preface it or delimit it to indicate that you’re switching from Markdown to HTML; you just use the tags.

TimMoore commented:
It looks like the GitHub behaviour of adding anchors for each title can be enabled in RedCarpet with the `with_toc_data` option (currently not enabled in Exercism).

kytrinyx commented:
The `with_toc_data` option sounds promising. @gusseting want to give it a try in this PR?

kytrinyx commented:
I'm going to go ahead and close this since it's been almost 3 weeks with no further movement. Feel free to reopen it if you come back to it!
